python/MoltresXS.py

```diff
@@ -0,0 +1,280 @@
+"""
+     Python Function that takes homoginezed cross sections and prepares them for MOLTRES
+
+"""
+
+import io
+import sys
+import numpy as np
+import h5py
+from pyne import serpent
+
+
+class scale_xs:
+    """
+        Python class that reads in a scale t16 file and organizes the cross section data
+        into a numpy dictionary. Currently set up to read an arbitrary number of energy groups,
+        an arbitrart number of delayed neutron groups, an arbitrary number of identities,
+        an arbitrary number of temperature branches, and an arbitrary number of burnups.
+
+        Parameters
+        ----------
+        FileIn: str
+            Name of file containing collapsed cross section data
+        Returns
+        ----------
+        xs_lib: dict
+            A hierarchical dictionary, organized by burnup, id, and temperature.
+            Currently stores REMXS, FISSXS, NSF, FISSE, DIFFCOEF, RECIPVEL, CHI, BETA_EFF, DECAY_CONSTANT and
+            GTRANSFXS.     
+    """
+
+    def __init__(self, FileIn):
+        with io.open(FileIn) as f:
+            lines = f.readlines()
+        struct = (lines[3].split()[:4])
+
+        num_burn, num_uni, num_groups = int(
+            struct[0]), int(struct[2]), int(struct[3])
+        ids = np.zeros(num_uni)
+        temps = np.array([])
+        burn = np.zeros([])
+        remxs = np.array([])
+        fissxs = np.array([])
+        nsf = np.array([])
+        fissE = np.array([])
+        difcoef = np.array([])
+        invV = np.array([])
+        chi = np.array([])
+        transxs = np.array([])
+        beta = np.array([])
+        decay = np.array([])
+
+        i = 0
+        k = 0
+        while k < len(lines):
+            line = lines[k]
+            if 'Identifier' in line:
+                val = int(lines[k+1].split()[0])
+                k += 1
+                if not any(val == ids):
+                    ids[i] = val
+                    i += 1
+            if 'Fuel temperatures' in line:
+                while 1:
+                    val = lines[k+1].split()
+                    k += 1
+                    for j in range(len(val)):
+                        temps = np.append(temps, float(val[j]))
+                    if "'" in lines[k+1]:
+                        break
+
+            if 'Nominal' in line:
+                val = lines[k+2].split()
+                k += 2
+                temps = np.append(temps, float(val[0]))
+            if 'Burnups' in line:
+                while 1:
+                    val = lines[k+1].split()
+                    k += 1
+                    for j in range(len(val)):
+                        burn = np.append(burn, float(val[j]))
+                    if "'" in lines[k+1]:
```

Comment:
Since you commonly use a "while 1 ... if x: break" type statement, why not consider a function like parseLine(str) that parses a SCALE-format line, then returns an array of values? I think this could cut down the length of this file significantly.